### PR TITLE
[ur] Adds usm testsuite

### PR DESCRIPTION
**Pull request by veselypeta**
Adds testsuite for USM testsuite to ur. Tests the following entrypoints

* `urMemFree`
* `urMemGetMemAllocInfo`
* `urUSMHostAlloc`
* `urUSMDeviceAlloc`
* `urUSMSharedAlloc`

There are some additional fixes to the spec as part of implementing this testsuite
- in `urMemGetMemAllocInfo` - `pPropValue` & `pPropValueSizeRet` arguments should be optional
- defined the `ur_usm_type_t` enumeration which was missing as the return type for the query of `UR_MEM_ALLOC_INFO_ALLOC_TYPE`

### CODE REVIEW

File: test/conformance/usm/urUSMSharedAlloc.cpp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
```
Comment (kbenzie) [inline test/conformance/usm/urUSMSharedAlloc.cpp:2]: This is missing files for testing `urMemFree` and `urMemGetMemAllocInfo` which are both int he [USM section](https://oneapi-src.github.io/unified-runtime/core/api.html#usm). There's perhaps some inconsistency with the naming of these entry points.

**Review by kbenzie — CHANGES_REQUESTED** [review]

File: test/conformance/usm/urUSMSharedAlloc.cpp (hunk 1 above)
Comment (veselypeta) [reply test/conformance/usm/urUSMSharedAlloc.cpp:2]: Yes I was confused by this slightly - I suppose they should be called `urUSMMemFree` & `urUSMGetMemAllocInfo`?

Comment (kbenzie) [reply test/conformance/usm/urUSMSharedAlloc.cpp:2]: Something along those lines, although the `M` in `USM` stands for memory so not including `Mem` would remove some reduncandy form the names. An entry point name change is deffo out of scope for this PR though.

Comment (veselypeta) [reply test/conformance/usm/urUSMSharedAlloc.cpp:2]: https://github.com/oneapi-src/unified-runtime/issues/201

**Review by kbenzie — APPROVED** [review]
LGTM

**Pull request merged**